controllers/velero.go

```diff
@@ -531,3 +530,31 @@ func (r *VeleroReconciler) getVeleroResourceReqs(velero *oadpv1alpha1.Velero) co
 
 	return ResourcesReqs
 }
+
+// For later: Move this code into validator.go when more need for validation arises
+// TODO: if multiple default plugins exist, ensure we validate all of them.
+// Right now its sequential validation
+func (r *VeleroReconciler) ValidateVeleroPiugins(log logr.Logger) (bool, error) {
+	velero := oadpv1alpha1.Velero{}
+	if err := r.Get(r.Context, r.NamespacedName, &velero); err != nil {
+		return false, err
+	}
+
+	secretName := ""
+	var defaultPlugin oadpv1alpha1.DefaultPlugin
+	for _, plugin := range velero.Spec.DefaultVeleroPlugins {
+		if pluginSpecificMap, ok := credentials.PluginSpecificFields[plugin]; ok && pluginSpecificMap.IsCloudProvider {
+			secretName = pluginSpecificMap.SecretName
```

Comment:
@savitharaghunathan Do you mind using the function specified here for getting the secret name, this function assumes the default name and key but also checks if the user has specified a different name and key for the particular provider https://github.com/openshift/oadp-operator/blob/7439793284b7112f22427478c5a275881fc04e2a/controllers/registry.go#L476

Reply:
@shubham-pampattiwar To get the `credential *corev1.SecretKeySelector` value, do I have to loop through bslspec? If so, then how it is different from the validations we are performing in the `bsl.go` ?

Reply:
@savitharaghunathan So there are 2 types of validations here based on the relationships:
1. _Cloud Provider Plugin - Secret Validation:_ This validation should make sure that there exists a credential secret if a cloud plugin is specified in the Velero CR.
2. _BSL - Secret Validation:_ This Validation should make sure that there exists a credential secret for an instance of BSL.

We already have the Second validation in place, but we do not have the first one. This PR should be implementing the first one, right ?

Reply:
Updated the code to reflect the changes we discussed yesterday.